eos-updater/poll.c

```diff
@@ -568,109 +558,23 @@ metadata_fetch_new (OstreeRepo    *repo,
-      g_autoptr(GAsyncResult) find_result = NULL, pull_result = NULL;
-      g_auto(GVariantBuilder) builder = G_VARIANT_BUILDER_INIT (G_VARIANT_TYPE ("a{sv}"));
-
-      g_debug ("%s: Finding remotes advertising upgrade_collection_ref: (%s, %s)",
-               G_STRFUNC, upgrade_collection_ref->collection_id, upgrade_collection_ref->ref_name);
-
-      ostree_repo_find_remotes_async (repo, refs, NULL  /* options */,
-                                      (OstreeRepoFinder **) finders->pdata,
-                                      NULL  /* progress */,
-                                      cancellable, async_result_cb, &find_result);
-
-      while (find_result == NULL)
-        g_main_context_iteration (context, TRUE);
-
-      g_clear_pointer (&results, ostree_repo_finder_result_freev);
-
-      results = ostree_repo_find_remotes_finish (repo, find_result, error);
-      if (results == NULL)
-        return NULL;
-
-      /* No updates available. */
-      if (results[0] == NULL)
-        {
-          g_message ("Poll: Couldn’t find any updates");
-          return NULL;
-        }
-
-      g_variant_builder_add (&builder, "{s@v}", "flags",
-                             g_variant_new_variant (g_variant_new_int32 (OSTREE_REPO_PULL_FLAGS_COMMIT_ONLY)));
-      pull_options = g_variant_ref_sink (g_variant_builder_end (&builder));
-
-      ostree_repo_pull_from_remotes_async (repo,
-                                           (const OstreeRepoFinderResult * const *) results,
-                                           pull_options, NULL  /* progress */, cancellable,
-                                           async_result_cb, &pull_result);
-
-      while (pull_result == NULL)
-        g_main_context_iteration (context, TRUE);
-
-      if (!ostree_repo_pull_from_remotes_finish (repo, pull_result, error))
-        return NULL;
-
-      g_clear_pointer (&checksum, g_free);
-      g_clear_pointer (&version, g_free);
-      g_clear_pointer (&new_refspec, g_free);
-      g_clear_pointer (&new_collection_ref, ostree_collection_ref_free);
-
-      /* Parse the commit and check there’s no redirection to a new ref. */
-      if (!parse_latest_commit (repo, upgrade_refspec, &redirect_followed,
-                                &checksum, &new_refspec, &new_collection_ref, &version,
-                                cancellable, error))
-        return NULL;
-
-      if (new_refspec != NULL)
-        {
-          g_clear_pointer (&upgrade_refspec, g_free);
-          upgrade_refspec = g_strdup (new_refspec);
-        }
-      if (new_collection_ref != NULL)
-        {
-          g_clear_pointer (&upgrade_collection_ref, ostree_collection_ref_free);
-          upgrade_collection_ref = g_steal_pointer (&new_collection_ref);
-        }
+      g_message ("Poll: Couldn’t find any updates");
+      return NULL;
     }
-  while (redirect_followed);
-
-  /* Final checks on the commit we found. */
-  if (!is_checksum_an_update (repo,
```

Comment:
So the `is_checksum_an_update()` checks are done as part of the call to `check_for_update_following_checkpoint_if_allowed()`, right?

Reply:
Right. In the case that no update was available you'll have `update_ref_info.commit == NULL` after the call to `check_for_update_following_checkpoint_if_allowed`.